src/main/java/com/twcable/grabbit/tools/monitor/RemoteJobStatusPoller.java

```diff
@@ -0,0 +1,88 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.twcable.grabbit.tools.monitor;
+
+import com.twcable.grabbit.tools.environment.Environment;
+import com.twcable.grabbit.tools.environment.UsernameAndPassword;
+import com.twcable.grabbit.tools.util.Utils;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.net.ConnectException;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+
+/**
+ * A {@link JobStatusPoller} that connects to a remote instance for job information.
+ */
+@SuppressWarnings("WeakerAccess")
+public class RemoteJobStatusPoller implements JobStatusPoller {
+    private final Environment environment;
+
+
+    public RemoteJobStatusPoller(Environment environment) {
+        this.environment = environment;
+    }
+
+
+    @Override
+    public JobStatus pollJobStatus(URI location, long jobId) {
+        final String jobStatusStr;
+        try {
+            final BufferedInputStream jobStatusStream = jobStatusOnClient(location, jobId);
+            jobStatusStr = Utils.toString(jobStatusStream);
+        }
+        catch (IOException e) {
+            throw Utils.softened(e);
+        }
+
+        // may be needed for psuedo-jobs ids like "all", but that's not used here...
+        // jobStatusStr = jobStatusStr.replaceFirst("\\[", "").replaceFirst("\\]", ""));
+
+        return JobStatus.fromJson(location, jobStatusStr);
+    }
+
+
+    @SuppressWarnings("PMD.PreserveStackTrace")
+    private BufferedInputStream jobStatusOnClient(URI baseUri, long jobId) throws IOException {
+        final URL url = new URL(baseUri.toURL(), "/grabbit/job/" + jobId + ".json");
+        try {
+            final HttpURLConnection httpCon = (HttpURLConnection)url.openConnection();
+            httpCon.setDoOutput(true);
+            httpCon.setRequestMethod("GET");
+            // httpCon.setRequestProperty("Content-Type", "application/json");
+            final UsernameAndPassword credentials = credentialsFor(baseUri);
+            httpCon.setRequestProperty("Authorization", "Basic " + credentials.basicAuthEncode());
+            return new BufferedInputStream(httpCon.getInputStream());
+        }
+        catch (ConnectException e) {
+            final ConnectException newExp = new ConnectException(e.getMessage() + " when trying to connect to " + url);
+            newExp.setStackTrace(e.getStackTrace());
+            throw newExp;
```

Comment:
Enriching the exception so that it's clear why it's complaining since the underlying `ConnectException` is really generic.